### ISSUE TITLE
pubsub: testing awspubsub in replay mode has a data race

### ISSUE THREAD

**Issue by ijt** [google/go-cloud#975]
The order of calls to ReceiveBatch and SendAcks is nondeterministic with internal/batcher using AddNoWait. The race goes away when the call to `s.ackBatcher.AddNoWait` is replaced with a blocking call to `s.ackBatcher.Add`.

One way to solve this problem is for `NewSubscription` to take its batcher as an argument, and for the test on AWS to pass in a batcher that blocks when adding items. There should be a separate PR to inject the batcher this way, and the PR for `awspubsub` (https://github.com/google/go-cloud/pull/953) should no longer fail on Travis once that is merged in and integrated.

**Comment by jba** [comment]
This should be fixed by a better replayer (see #770). I'll make that a priority for the next sprint.

**Comment by ijt** [comment]
Okay, handing this issue off to you.

**Comment by vangent** [comment]
Per discussion in standup, a better replayer won't resolve this.

**Comment by ijt** [comment]
Being able to pass in an ack batcher means this won't be a problem anymore: a5b87531. Closing it.
